packages/lib/entityPermissionUtils.ts

```diff
@@ -19,6 +22,18 @@ export function canEditEntity(
   );
 }
 
+export function canAccessEntity(
+  entity: Parameters<typeof getEntityPermissionLevel>[0],
+  userId: Parameters<typeof getEntityPermissionLevel>[1]
+) {
+  const permissionLevel = getEntityPermissionLevel(entity, userId);
+  return (
+    permissionLevel === ENTITY_PERMISSION_LEVEL.TEAM_WRITE ||
+    permissionLevel === ENTITY_PERMISSION_LEVEL.USER_ONLY_WRITE ||
+    permissionLevel === ENTITY_PERMISSION_LEVEL.TEAM_READ_ONLY
+  );
+}
+
```

Comment:
We can optimize `getEntityPermissionLevel` function but not fetching all the members of team and passing it to entity. We can fetch the membership directly from Table